ipywidgets/widgets/widget_string.py

```diff
@@ -107,3 +107,40 @@ def on_submit(self, callback, remove=False):
             Whether to unregister the callback
         """
         self._submission_callbacks.register_callback(callback, remove=remove)
+
+
+@register
+class Password(_String):
```

Comment:
Would it be easier to just inherit from Text?

```
@register
class Password(Text):
    _view_name = ...
    _model_name = ...
```
Then you inherit all of the text behavior automatically.